### DISCUSSION TITLE
How to handle nullable decimal in entity file?

### DISCUSSION THREAD

**Discussion by yzl2250** [mikro-orm/mikro-orm#3100] [Q&A]
Hi,

I'm getting the following error when I try to query a table that consists of a nullable decimal field (MySql)
"Trying to set Object.costing of type 'number' to '10.20' of type 'string'",

Is there any option to handle it in the entity file? 
`    @Property()
    costing?: number;`

Btw, I am using _TsMorphMetadataProvider_.

Thanks.

**Comment by yzl2250** [comment]
Sorry my mistake, just have to change from getResult() to execute('get').
Thanks.

**Comment by B4nan** [comment]
This is wrong, decimals are mapped to string, as you could lost precision, js numbers cant hold the same information. If you want them to be mapped to numbers, you need to create custom type for that. 

Your error message also points to using POJO instead of entity instance, which is wrong as well. Hard to say why, as you havent provided any actual code.

**Reply by yzl2250** [reply]
I'm generally new to js/ts and mikro-orm, would appreciate your comment on what should be done instead.
Please let me know if I miss out any info, much thanks.

```
@Entity()
export class Event {

    @PrimaryKey()
    id: number;

    @Property()
    costing?: number
}
```


```
@Injectable()
export class EventService {
    constructor(
        @InjectRepository(Event) 
        private readonly eventRepository: EntityRepository<Event>,
   }

  async getEvent() {
    const qb = this.eventRepository.createQueryBuilder('event')
          .select(['event.id', 'event.costing']);
    return await qb.execute('all');
  }
}
```
